plugins/by-name/dap/default.nix

```diff
@@ -22,10 +22,24 @@ lib.nixvim.plugins.mkNeovimPlugin {
   deprecateExtraOptions = true;
 
   extraOptions = {
-    adapters = lib.nixvim.mkCompositeOption "Dap adapters." {
-      executables = dapHelpers.mkAdapterOption "executable" dapHelpers.executableAdapterOption;
-      servers = dapHelpers.mkAdapterOption "server" dapHelpers.serverAdapterOption;
-      pipes = dapHelpers.mkAdapterOption "pipe" dapHelpers.pipeAdapterOption;
+    adapters = lib.nixvim.mkNullOrOption' {
+      type = types.submodule {
+        freeformType = types.attrsOf types.rawLua;
```

Comment:
I assume we don't need `adapters = mkRaw ""` support?

This freeformType will allow defining non-option attrs as rawLua, but it will not allow defining explicitly declared options as rawLua (unless the option's type supports it) and it will not allow defining rawLua instead of the root submodule.

That may be what you intended, I'm just double checking.